### PR TITLE
Add convenience context menus for RunContext and oLibrary tagging

### PR DESCRIPTION
**Pull request by unix-system**
## Overview

Adds the following commands:

- `overture-vscode.toggleLibrary` - Adds or removes the `oLibrary` tag from the selected Script. Available via Context Menu.
- `overture-vscode.setServerContext` - Sets the selected Script's RunContext to Server. Available via Context Menu.
- `overture-vscode.setClientContext` - Sets the selected Script's RunContext to Client. Available via Context Menu.

Adds the following configuration settings:

- `overture-vscode.addRunContextMenu` *(defaults true)* - Should the "Mark Script with ... RunContext" explorer menu be shown on ".server.lua(u)" and ".client.lua(u)" files.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,33 +9,83 @@
...
 	"categories": [
-		"Snippets", "Programming Languages", "Other"
+		"Snippets",
+		"Programming Languages",
+		"Other"
+	],
+	"activationEvents": [
+		"onStartupFinished"
 	],
-	"activationEvents": ["onStartupFinished"],
```
Comment (devSparkle) [inline package.json:19]: Curious removal. Are there any implications?

I distinctly recall this fixing a particular bug.

Comment (devSparkle) [reply package.json:19]: Actually, here is exactly why this is needed: https://stackoverflow.com/questions/56086972/how-do-i-run-an-extension-command-on-startup-on-vscode

This trick is used to actually initialise the extension's configuration of settings.

**Review by devSparkle — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (unix-system) [reply package.json:19]: Oh, not sure why I actually did this. Rolling back.

Comment (unix-system) [reply package.json:19]: Nevermind. It's just indented 🤣 
image

**Review by devSparkle — APPROVED** [review]

**Pull request merged**